go-controller/observability-lib/sampledecoder/sample_decoder.go

```diff
@@ -133,90 +134,76 @@ func (d *SampleDecoder) DecodeCookieIDs(obsDomainID, obsPointID uint32) (string,
-		default:
-			action = "Action " + o.Action
-		}
-		actor := o.ExternalIDs[libovsdbops.OwnerTypeKey.String()]
-		var msg string
-		switch actor {
-		case libovsdbops.AdminNetworkPolicyOwnerType:
-			msg = fmt.Sprintf("admin network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.BaselineAdminNetworkPolicyOwnerType:
-			msg = fmt.Sprintf("baseline admin network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastNamespaceOwnerType:
-			msg = fmt.Sprintf("multicast in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastClusterOwnerType:
-			msg = fmt.Sprintf("cluster multicast policy, direction %s", o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNodeOwnerType:
-			msg = "default allow from local node policy, direction ingress"
-		case libovsdbops.NetworkPolicyOwnerType:
-			msg = fmt.Sprintf("network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNamespaceOwnerType:
-			msg = fmt.Sprintf("network policies isolation in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.EgressFirewallOwnerType:
-			msg = fmt.Sprintf("egress firewall in namespace %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
-		case libovsdbops.UDNIsolationOwnerType:
-			msg = fmt.Sprintf("UDN isolation of type %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
-		}
-		return fmt.Sprintf("%s by %s", action, msg)
-	default:
-		return ""
+func newACLEvent(o *nbdb.ACL) *model.ACLEvent {
+	actor := o.ExternalIDs[libovsdbops.OwnerTypeKey.String()]
+	event := model.ACLEvent{
+		Action: o.Action,
+		Actor:  actor,
+	}
+	switch actor {
+	case libovsdbops.AdminNetworkPolicyOwnerType, libovsdbops.BaselineAdminNetworkPolicyOwnerType, libovsdbops.NetworkPolicyOwnerType:
+		event.Name = o.ExternalIDs[libovsdbops.ObjectNameKey.String()]
+		event.Direction = o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()]
+	case libovsdbops.MulticastNamespaceOwnerType, libovsdbops.NetpolNamespaceOwnerType:
+		event.Namespace = o.ExternalIDs[libovsdbops.ObjectNameKey.String()]
+		event.Direction = o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()]
+	case libovsdbops.MulticastClusterOwnerType:
+		event.Direction = o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()]
+	case libovsdbops.EgressFirewallOwnerType:
+		event.Namespace = o.ExternalIDs[libovsdbops.ObjectNameKey.String()]
+		event.Direction = "Egress"
+	case libovsdbops.UDNIsolationOwnerType:
+		event.Name = o.ExternalIDs[libovsdbops.ObjectNameKey.String()]
```

Comment:
this function doesn't handle `NetpolNodeOwnerType` I think u can statically populate name, action and direction

Reply:
yes it does, in the first case: 
```golang
case libovsdbops.AdminNetworkPolicyOwnerType, libovsdbops.BaselineAdminNetworkPolicyOwnerType, libovsdbops.NetworkPolicyOwnerType
```

Reply:
In fact `NetpolNodeOwnerType` should work and has Action and Actor already populated (see lines 169-170), it just that it hasn't got other extra variables. I can add direction indeed.